src/Registries/RouteRegistry.php

```diff
@@ -25,7 +25,7 @@ public function __construct(Router $router)
     /**
      * @param $route
      */
-    public function register($route)
+    public function register($route, $shared = false)
```

Comment:
This is bad practice. You should follow YAGNI principle. Either create a new method called `registerShared` or get the `shared` variable with setters and getters. `$this->routeRegistry->shared($value = true)->register()`